### PR TITLE
Feature/BP-966 second chance clean

### PR DESCRIPTION
**Pull request by serpentscode**
BP-966 Cleanup database tables for second chance functionality.

### CODE REVIEW

File: Cron/SecondChancePrune.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * @var \Buckaroo\Magento2\Model\SecondChanceRepository
+     */
+    protected $secondChanceRepository;
+
+    /**
+     * @param \Buckaroo\Magento2\Model\SecondChanceRepository       $secondChanceRepository
+     */
+    public function __construct(
+        \Buckaroo\Magento2\Model\SecondChanceRepository $secondChanceRepository
```
Comment (florinel-chis) [inline Cron/SecondChancePrune.php:33]: move to top (use ..) and use the short class name

File: etc/adminhtml/system/account.xml [hunk 2]
```diff
@@ -311,6 +311,15 @@
                 </depends>
             </field>
```
Comment (florinel-chis) [inline etc/adminhtml/system/account.xml:313]: scope is global, therefore show in website and store view - should not be present

File: Model/SecondChanceRepository.php [hunk 3]
```diff
@@ -292,12 +292,37 @@ public function deleteById($secondChanceId)
...
+
+        if ($days <= 0) {
+            return false;
+        }
+
+        $connection = $this->resource->getConnection();
+        try {
+            $connection->delete(
+                $this->resource->getMainTable(),
+                ['created_at < (NOW() - INTERVAL ? DAY)' => $days]
```
Comment (florinel-chis) [inline Model/SecondChanceRepository.php:311]: lt instead if <

Comment (serpentscode) [reply Model/SecondChanceRepository.php]: fixed!

**Review by florinel-chis — DISMISSED** [review]

**Comment by vladislav-padalka-hys** [comment]
@serpentscode assuming necessity of this functionality to be included to 1.37.0 release - please merge it to https://github.com/buckaroo-it/Magento2/tree/develop and to https://github.com/buckaroo-it/Magento2/tree/releases/1.37.0

**Comment by saranchuk-hys** [comment]
Hello @florinel-chis , @serpentscode sorry but i see missing missing important things:
1) Scope is global, but other Second Chance options allowed work for multistore, same as SC table (contains site_id field), so i think will be better provide multistore options and for this feature.
2) Label "Prune older second chance entries (days)" - not informative, it's not explain what is exactly will be deleted and what heritage is (links in already sended SC email will not works anymore).
3) Based on previos, Merchants will start complaint what links go to emty cart, so will be greate to show user some message "This token not working anymore, please continue your shoping"

**Comment by florinel-chis** [comment]
@saranchuk-hys 
1) agree, that needs to be fixed.
2) that is descriptive enough and correct.
3) that is not in scope of this ticket.

File: Setup/UpgradeSchema.php [hunk 4]
```diff
@@ -411,5 +411,11 @@ protected function createOptimizationIndexes(SchemaSetupInterface $installer)
             $installer->getIdxName('sales_payment_transaction', ['txn_id']),
             ['txn_id']
         );
+
+        $installer->getConnection()->addIndex(
+            $installer->getTable('buckaroo_magento2_second_chance'),
+            $installer->getIdxName('buckaroo_magento2_second_chance', ['store_id']),
```
Comment (saranchuk-hys) [inline Setup/UpgradeSchema.php:417]: Hello, for new indexes we use https://github.com/buckaroo-it/Magento2/blob/develop/etc/db_schema.xml#L48 , will be great if you use as well

Comment (serpentscode) [reply Setup/UpgradeSchema.php:417]: Hi there, thanks for the info, I'll do the changes.

**Review by florinel-chis — APPROVED** [review]

**Pull request merged**